### ISSUE TITLE
Add warning when ContainsConstraint is used against incompatible argument type

### ISSUE THREAD

**Issue by Dreamescaper** [nunit/nunit.analyzers#221]
```csharp
var actual = new [] {"1", "2", "3"};
var expected = 2;
Assert.That(actual, Does.Contain(expected));   <-- warning expected
```

actual should be either string or IEnumerable.
If actual is string, expected should be string as well.
If expected is IEnumerable<T>, expected should have matching with T type (using rules that we have in EqualToIncompatibleTypesAnalyzer.CanBeAssertedForEquality)
